### PR TITLE
[bsp/bluetrum] add ble support

### PR DESCRIPTION
**Pull request by greedyhao**
## 拉取/合并请求描述：(PR description)

[
添加 ble 支持
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [x] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [ ] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [ ] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/coding_style_en.txt)

### CODE REVIEW

File: bsp/bluetrum/ab32vg1-ab-prougen/link.lds [hunk 1]
```diff
@@ -21,6 +23,7 @@ MEMORY
     flash(rx)       : org = __base + 512,       len = __max_flash_size
     comm(rx)        : org = __comm_vma,         len = __comm_ram_size
 
+    bram			: org = __bram_vma,         len = __ble_ram_size
```
Comment (mysterywolf) [inline bsp/bluetrum/ab32vg1-ab-prougen/link.lds:26]: 要用空格代替tab

File: bsp/bluetrum/ab32vg1-ab-prougen/board/board.c [hunk 2]
```diff
@@ -145,6 +145,11 @@ void rt_hw_board_init(void)
 #endif
 }
 
+void rt_hw_us_delay(rt_uint32_t us)
+{
+
+}
+
```
Comment (mysterywolf) [inline bsp/bluetrum/ab32vg1-ab-prougen/board/board.c:152]: rt_hw_us_delay可以不用加了，在delay.c里有一个weak版本的

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: bsp/bluetrum/ab32vg1-ab-prougen/board/board.c (hunk 2 above)
Comment (greedyhao) [reply bsp/bluetrum/ab32vg1-ab-prougen/board/board.c]: 已改

File: bsp/bluetrum/ab32vg1-ab-prougen/applications/blehr_app.c [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+#define nsmem_cb_init   ((nsmem_cb_init_func)0x84140)
+
+int btctrl_mem_init(void)
+{
+    void *heap_buf;
+    uint16_t *heap_size_buf;
+    uint32_t block_size;
+
+    bthw_get_heap_info(&heap_buf, &heap_size_buf, &block_size);
+    // rt_kprintf("heap_buf=0x%p heap_size_buf=0x%p block_size=0x%p\n", heap_buf, heap_size_buf, block_size);
```
Comment (Guozhanxin) [inline bsp/bluetrum/ab32vg1-ab-prougen/applications/blehr_app.c:36]: 请删除注释掉的代码或改为 LOG_D

Comment (greedyhao) [reply bsp/bluetrum/ab32vg1-ab-prougen/applications/blehr_app.c]: 已改

**Pull request merged**